doc/system-telemetry/reboot-cause.md

```diff
@@ -0,0 +1,150 @@
+# Reboot-cause information via telemetry agent
+
+## Revision
+
+| Rev | Date     | Author      | Change Description |
+|:---:|:--------:|:-----------:|--------------------|
+| 0.1 | 09/02/20 | Sujin Kang  | Initial version    |
+
+## Scope
+-Enable sonic streaming telemetry agent to send Reboot-cause information
+
+### Enable sonic streaming telemetry agent to send Reboot-cause information
+
+#### Part 1
+During the boot, the `determine-reboot-cause` service ( previously `process-reboot-cause`) determines the last reboot-cause based on the hardware reboot-cause
+and the software reboot-cause information and `determine-reboot-cause` service will save the formatted last previous
+reboot cause information to "/host/reboot-cause/previous-reboot-cause/" with adding timestamp at the end of file name.
+`determine-reboot-cause` also will also create a symbolic link of the last reboot cause file to "/host/reboot-cause/previous-reboot-cause.txt"
+
+The example shows the previous reboot-cause files stored in /host/reboot-cause/previous-reboot-cause/.
+```
+admin@sonic:~$ ls /host/reboot-cause/previous-reboot-cause/
+previous-reboot-cause-2020_10_09_01_56_59.txt
+previous-reboot-cause-2020_10_09_02_00_53.txt
+previous-reboot-cause-2020_10_09_02_33_06.txt
+previous-reboot-cause-2020_10_09_04_53_58.txt
```

Comment:
Suggest removing the "previous-" prefix from these filenames. "previous-" was used to differentiate the previous reboot cause from the upcoming reboot, if it was software-initiated, but now we have a different directory structure.

Reply:
Also, why do these files end in a `.txt` suffix, since they contain JSON, not plain text?